crates/astria-sequencer/src/mempool.rs

```diff
@@ -157,23 +255,33 @@ impl Mempool {
         let enqueued_tx = EnqueuedTransaction::new(tx);
         let priority = enqueued_tx.priority(current_account_nonce)?;
         let tx_hash = enqueued_tx.tx_hash;
-        self.inner.write().await.push(enqueued_tx, priority);
-        tracing::trace!(
-            tx_hash = %telemetry::display::hex(&tx_hash),
-            "inserted transaction into mempool"
-        );
+
+        let mut queue = self.queue.write().await;
+
+        if queue.get_priority(&enqueued_tx).is_some() {
+            // already exists, update nonce priority but preserve timestamp
+            queue.change_priority(&enqueued_tx, priority);
+        } else {
+            // doesn't exist
+            queue.push(enqueued_tx, priority);
+            tracing::trace!(
+                tx_hash = %telemetry::display::hex(&tx_hash),
+                "inserted transaction into mempool"
+            );
+        }
+
         Ok(())
     }
 
     /// inserts all the given transactions into the mempool
     pub(crate) async fn insert_all(&self, txs: Vec<(EnqueuedTransaction, TransactionPriority)>) {
-        self.inner.write().await.extend(txs);
+        self.queue.write().await.extend(txs);
```

Comment:
this isn't the same behaviour as `insert` as it doesn't preserve the timestamp if it already exists

Reply:
You're right. I was just relying on the behavior that we're currently only using this function to re-insert transactions that we just popped. Do you have a preference of me adding a comment that explains that or to do a similar process as in `insert()`?

Reply:
i think doing a similar process as to `insert()` makes the most sense!